jerry-ext/include/jerryscript-ext/handle-scope.h

```diff
@@ -0,0 +1,129 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRYX_HANDLE_SCOPE_H
+#define JERRYX_HANDLE_SCOPE_H
+
+#include "jerryscript.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif /* __cplusplus */
+
+#ifndef JERRYX_HANDLE_PRELIST_SIZE
+#define JERRYX_HANDLE_PRELIST_SIZE 20
+#endif
+
+#ifndef JERRYX_SCOPE_PRELIST_SIZE
+#define JERRYX_SCOPE_PRELIST_SIZE 20
+#endif
+
+typedef struct jerryx_handle_t jerryx_handle_t;
+/**
+ * Dynamically allocated handle in the scopes.
+ * Scopes has it's own size-limited linear storage of handles. Still there
+ * might be not enough space left for new handles, dynamically allocated
+ * `jerryx_handle_t` could ease the pre-allocated linear memory burden.
+ */
+struct jerryx_handle_t
+{
+  jerry_value_t jval; /**< jerry value of the handle bound to */
+  jerryx_handle_t *sibling; /**< next sibling the the handle */
+};
+
+#define JERRYX_HANDLE_SCOPE_FIELDS                          \
+  jerry_value_t handle_prelist[JERRYX_HANDLE_PRELIST_SIZE]; \
+  size_t handle_count;                                      \
```

Comment:
Am I understand correctly that handle_count must be lass than JERRYX_HANDLE_PRELIST_SIZE? If so, uint8_t should be enough.

Reply:
No, `handle_count` could grows greater than JERRYX_HANDLE_PRELIST_SIZE, after then new handles would be allocated outside of prelist as a chained list instead.

Reply:
Do you need to count those handles in the chained list? It seems to me there is no api function to get the number of handles, and the rest of the functions could be changed to handle_count to not grow after reaching JERRYX_HANDLE_PRELIST_SIZE. Is there a function where it would be a large performance drawback to do this?

Reply:
Updated :). From now on only handles in prelist would be counted.